mitiq/calibration/settings.py

```diff
@@ -0,0 +1,137 @@
+# Copyright (C) 2021 Unitary Fund
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+from dataclasses import dataclass
+from functools import partial
+from itertools import product
+from typing import Any, Callable
+
+import networkx as nx
+
+from mitiq._typing import QPROGRAM
+from mitiq.benchmarks import (
+    generate_ghz_circuit,
+    generate_mirror_circuit,
+    generate_quantum_volume_circuit,
+    generate_rb_circuits,
+)
+from mitiq.pec import execute_with_pec
+from mitiq.zne import execute_with_zne
+from mitiq.zne.inference import ExpFactory, LinearFactory, RichardsonFactory
+from mitiq.zne.scaling import (
+    fold_gates_at_random,
+    fold_global,
+    insert_id_layers,
+)
+
+
+@dataclass
+class CircuitData:
+    circuit: QPROGRAM
+    dimensions: tuple[int, int]
+    type: str
+    two_qubit_gate_count: int
+
+
+class Settings:
+    """A class to store settings relating to error mitigation calibration."""
+
+    def __init__(
+        self,
+        mitigation_methods: list[str],
+        circuit_types: list[str],
+        circuit_dimensions: tuple[int, int],
+        method_params: dict[str, Any],
+    ):
+        self.mitigation_methods = mitigation_methods
+        self.method_params = method_params
+        self.circuit_types = circuit_types
+        self.circuit_dimensions = circuit_dimensions
+
+    def make_circuits(self) -> list[CircuitData]:
```

Comment:
This function should also save the correct bitstring for each benchmark circuits. E.g. in the `ideal_result` attribute of `CircuitData`.